### PR TITLE
Re-add LinearAlgebra::set_zero_mean_value().

### PR DESCRIPTION
**Pull request by bangerth**
Addresses the discussion in #15688.

### CODE REVIEW

File: include/deal.II/lac/vector_space_vector.h [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+namespace LinearAlgebra
+{
+  /**
+   * Shift all entries of the vector by a constant factor so that the mean
+   * value of the vector becomes zero.
+   */
+  template <typename VectorType>
+  void
+  set_zero_mean_value(VectorType &vector)
```
Comment (kronbichler) [inline include/deal.II/lac/vector_space_vector.h:36]: Did we agree on deprecating this feature, as suggested by https://github.com/dealii/dealii/pull/15688#discussion_r1258201969?
```suggestion
  template <typename VectorType>
  DEAL_II_DEPRECATED void
  set_zero_mean_value(VectorType &vector)
```

Comment (bangerth) [reply include/deal.II/lac/vector_space_vector.h:36]: Actually, while I dislike the name of the function, I think `lac/` is the right place for it and `VectorTools` is not. The latter deals with finite element fields, which are defined on a mesh and with regard to a `DoFHandler`. This function is defined on a purely algebraic vector level. I think this is the right directory for it, if perhaps not the right file.

**Review by tamiko — APPROVED** [review]

File: include/deal.II/lac/vector_space_vector.h (hunk 1 above)
Comment (tamiko) [reply include/deal.II/lac/vector_space_vector.h:36]: @kronbichler @bangerth *ping*

Comment (masterleinad) [reply include/deal.II/lac/vector_space_vector.h:36]: I think `DEPRECATED_EARLY` is our standard now. We also need to print a warning when including this file.

**Review by tamiko — DISMISSED** [review]
As @masterleinad pointed out, we should use `DEAL_II_DEPRECATED_EARLY` and also use an include `#warning` for the header.

**Comment by bangerth** [comment]
As I mentioned in one of the comments, I don't actually want to deprecate the function.

**Comment by tamiko** [comment]
@bangerth Then let's find a better header for it.

**Comment by bangerth** [comment]
How about we merge the current PR, and then I'll write a separate one in which I deprecate this function and provide it elsewhere under a better name?

**Comment by tamiko** [comment]
@bangerth works for me.

**Review by tamiko — APPROVED** [review]

**Pull request merged**